src/uattributes/uattributesvalidator.rs

```diff
@@ -360,193 +250,253 @@ impl UAttributesValidators {
+            Ok(())
+        } else {
+            Err(UAttributesError::validation_error(error_message))
         }
-        self.validate_source(attributes)
-            .and_then(|_| self.validate_sink(attributes))
     }
 }
 
 /// Validate `UAttributes` with type `UMessageType::Request`
 pub struct RequestValidator;
 
-impl UAttributesValidator for RequestValidator {
-    fn type_name(&self) -> &'static str {
-        "UAttributesValidator.Request"
+impl RequestValidator {
+    /// Verifies that a set of attributes contains a valid permission level.
+    ///
+    /// # Errors
+    ///
+    /// If [`UAttributes::permission_level`] contains a value that is less than 1, an error is returned.
+    pub fn validate_permission_level(
+        &self,
+        attributes: &UAttributes,
+    ) -> Result<(), UAttributesError> {
+        if let Some(plevel) = attributes.permission_level {
+            if plevel < 1 {
+                return Err(UAttributesError::validation_error(format!(
+                    "Permission Level must be a positive integer [{plevel}]"
+                )));
+            }
+        }
+        Ok(())
     }
+}
 
-    fn requires_request_id(&self) -> bool {
-        true
+impl UAttributesValidator for RequestValidator {
+    fn message_type(&self) -> UMessageType {
+        UMessageType::UMESSAGE_TYPE_REQUEST
     }
 
-    /// Validates that attributes for a message meant for an RPC request has the correct type.
-    ///
-    /// # Arguments
+    /// Checks if a given set of attributes complies with the rules specified for
+    /// RPC request messages.
     ///
-    /// * `attributes` - `UAttributes` object containing the message type to validate.
+    /// # Errors
     ///
-    /// # Returns
+    /// Returns an error if any of the following checks fail for the given attributes:
     ///
-    /// Returns a `ValidationResult` that is success or failed with a failure message.
-    fn validate_type(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
-        match attributes.type_.enum_value() {
-            Err(unknown_code) => Err(UAttributesError::validation_error(format!(
-                "Unknown Attribute Type [{}]",
-                unknown_code
-            ))),
-            Ok(mt) => match mt {
-                UMessageType::UMESSAGE_TYPE_REQUEST => Ok(()),
-                _ => Err(UAttributesError::validation_error(format!(
-                    "Wrong Attribute Type [{}]",
-                    mt.to_type_string()
-                ))),
-            },
+    /// * [`UAttributesValidator::validate_type`]
+    /// * [`UAttributesValidator::validate_id`]
+    /// * [`UAttributesValidator::validate_ttl`]
+    /// * [`UAttributesValidator::validate_source`]
+    /// * [`UAttributesValidator::validate_sink`]
+    /// * [`RequestValidator::validate_permission_level`]
+    fn validate(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
+        let error_message = vec![
```

Comment:
Nice! So we can return multiple different things that "went wrong". :slightly_smiling_face:

Reply:
I also like this idea.